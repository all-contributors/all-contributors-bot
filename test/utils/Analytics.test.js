const nock = require('nock')

const Analytics = require('../../src/utils/Analytics')

describe('Analytics', () => {
    test('Analytics', async () => {
        const analytics = new Analytics({
            repo: 'all-contributors-bot',
            owner: 'all-contributors',
            user: 'mockusername',
            apiKey: 'mock api key',
        })

        nock('https://api.amplitude.com')
            .post(`/httpapi`, body => {
                expect(body).toMatchSnapshot()
                return true
            })
            .reply(200)

        analytics.track('my-event')

        await analytics.finishQueue()
    })
})
