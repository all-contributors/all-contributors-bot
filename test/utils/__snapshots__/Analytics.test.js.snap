// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Analytics 1`] = `
Object {
  "api_key": "mock api key",
  "event": Array [
    Object {
      "event_properties": Object {
        "funnel_id": "mockFunnelId",
      },
      "event_type": "my-event",
      "user_id": "mockusername",
      "user_properties": Object {
        "owner": "all-contributors",
        "repo": "all-contributors-bot",
      },
    },
  ],
}
`;
